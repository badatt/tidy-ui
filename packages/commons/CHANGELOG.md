# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.0.43](https://github.com/badatt/tidy-ui/compare/v0.0.42...v0.0.43) (2022-07-22)

**Note:** Version bump only for package @tidy-ui/commons





## [0.0.42](https://github.com/badatt/tidy-ui/compare/v0.0.41...v0.0.42) (2022-07-17)

**Note:** Version bump only for package @tidy-ui/commons





## [0.0.41](https://github.com/badatt/tidy-ui/compare/v0.0.40...v0.0.41) (2022-07-17)

**Note:** Version bump only for package @tidy-ui/commons





## [0.0.40](https://github.com/badatt/tidy-ui/compare/v0.0.39...v0.0.40) (2022-07-17)


### Bug Fixes

* view box fix ([7fd569a](https://github.com/badatt/tidy-ui/commit/7fd569afb88ed03b70bfe3ce4c97f927954f3351))


### Features

* message widget other features and cleanup ([e5329b2](https://github.com/badatt/tidy-ui/commit/e5329b288365daeef1f126423fb94ee4076788f0))





## [0.0.39](https://github.com/badatt/tidy-ui/compare/v0.0.38...v0.0.39) (2022-07-14)

**Note:** Version bump only for package @tidy-ui/commons
