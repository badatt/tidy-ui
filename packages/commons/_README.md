#include "../../internals/md/banner-logo.md"

<h1 align="center">@tidy-ui/commons</h1>

#include "../../internals/md/checks.md"
