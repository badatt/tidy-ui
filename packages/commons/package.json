{
  "name": "@tidy-ui/commons",
  "version": "0.0.37",
  "scripts": {
    "build": "yarn clean && rollup -c ../../rollup.config.js --environment PACKAGE_NAME:commons",
    "clean": "tsc --build --clean && rimraf -rf dist && rimraf -rf build",
    "format": "prettier \"**/*.{js,jsx,json,yml,yaml,css,less,scss,ts,tsx,md,graphql,mdx}\" --write",
    "lint": "eslint -c ../../.eslintrc --ext .js,.jsx,.ts,.tsx src",
    "lint-fix": "yarn format && eslint -c ../../.eslintrc --ext .js,.jsx,.ts,.tsx src --fix",
    "postinstall": "rm -rf node_modules/@types/react-native",
    "reset": "yarn clean && rimraf -rf node_modules",
    "test": "yarn \"test:coverage\"",
    "test:coverage": "jest --bail --coverage --passWithNoTests",
    "test:update": "jest --bail --coverage --passWithNoTests -u",
    "test:watch": "jest --watchAll --verbose",
    "vet": "yarn lint",
    "yalc:publish": "yarn build && yalc publish",
    "yalc:push": "yarn build && yalc push"
  },
  "peerDependencies": {
    "react": "^16.0.0",
    "styled-components": "^5.3.3"
  },
  "devDependencies": {
    "@rollup/plugin-commonjs": "^21.0.1",
    "@rollup/plugin-node-resolve": "^13.1.1",
    "@rollup/plugin-replace": "^3.0.1",
    "@types/react": "^16.0.0",
    "@types/styled-components": "^5.1.17",
    "@typescript-eslint/eslint-plugin": "^5.4.0",
    "@typescript-eslint/parser": "^5.4.0",
    "eslint": "^8.3.0",
    "is-ci-cli": "^2.2.0",
    "react": "^16.0.0",
    "rollup": "^2.61.1",
    "rollup-plugin-json": "^4.0.0",
    "rollup-plugin-node-polyfills": "^0.2.1",
    "rollup-plugin-terser": "^7.0.2",
    "rollup-plugin-typescript2": "^0.31.1",
    "typescript": "^4.4.4"
  },
  "main": "dist/index.cjs.js",
  "module": "dist/index.es.js",
  "typings": "dist/src/index.d.ts",
  "types": "dist/src/index.d.ts",
  "license": "MIT",
  "description": "tidy-ui react component library; 'commons' package",
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "dist"
  ]
}
