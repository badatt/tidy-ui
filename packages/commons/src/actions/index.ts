import { PageActionTypes } from './page.actions';
import { RootActionTypes } from './root.actions';

/**
 * Tidy UI components internal actions
 */
export type AppActionTypes = PageActionTypes | RootActionTypes;

export { PageActions } from './page.actions';
export { RootActions } from './root.actions';
