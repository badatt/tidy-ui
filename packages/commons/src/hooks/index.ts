export { default as useColor } from './useColor';
export { default as useEscaper } from './useEscaper';
export { default as useHash } from './useHash';
export { default as useIsMounted } from './useIsMounted';
export { default as useKeyDown } from './useKeyDown';
export { default as useKeyPress } from './useKeyPress';
export { default as usePortal } from './usePortal';
export { default as useTheme } from './useTheme';
export { default as useTimeout } from './useTimeout';
export { default as useWindow } from './useWindow';
