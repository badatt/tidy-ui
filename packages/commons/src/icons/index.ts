export { default as AddCircleIcon } from './AddCircleIcon';
export { default as AddIcon } from './AddIcon';
export { default as CancelIcon } from './CancelIcon';
export { default as CheckCircleIcon } from './CheckCircleIcon';
export { default as CloseIcon } from './CloseIcon';
export { default as DangerousIcon } from './DangerousIcon';
export { default as ExpandLessIcon } from './ExpandLessIcon';
export { default as ExpandMoreIcon } from './ExpandMoreIcon';
export * from './Icon';
export { default as InfoIcon } from './InfoIcon';
export { default as LaunchIcon } from './LaunchIcon';
export { default as LinkIcon } from './LinkIcon';
export { default as RemoveIcon } from './RemoveIcon';
export { default as RotatingCircleIcon } from './RotatingCircleIcon';
export { default as WarningIcon } from './WarningIcon';
