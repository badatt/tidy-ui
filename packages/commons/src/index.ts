export * from './actions';
export * from './hooks';
export * from './icons';
export * from './provider';
export * from './reducers';
export * from './theme';
export * from './utils';
