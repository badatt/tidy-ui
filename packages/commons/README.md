<p align="center">
  <img width="250" src="https://raw.githubusercontent.com/badatt/tidy-ui/main/internals/docs/tidi-ui-logo-320X320.png" alt="Tidy UI" />
</p>


<h1 align="center">
  @tidy-ui/commons
  <a href="https://www.npmjs.com/package/@tidy-ui/commons">
    <img alt="npm (scoped)" src="https://img.shields.io/npm/v/@tidy-ui/commons" />
  </a>
</h1>
<p align="center">
  <a href="https://sonarcloud.io/summary/new_code?id=badatt_tidy-ui">
    <img
      src="https://sonarcloud.io/api/project_badges/measure?project=badatt_tidy-ui&metric=alert_status"
      alt="Quality Gate Status"
    />
  </a>
</p>
<p align="center">
  <a href="https://github.com/badatt/tidy-ui/actions/workflows/publish-wf.yml">
    <img src="https://github.com/badatt/tidy-ui/actions/workflows/publish-wf.yml/badge.svg" alt="Publishing" />
  </a>
  <a href="https://sonarcloud.io/summary/new_code?id=badatt_tidy-ui">
    <img src="https://sonarcloud.io/api/project_badges/measure?project=badatt_tidy-ui&metric=coverage" alt="Coverage" />
  </a>
  <a href="https://sonarcloud.io/summary/new_code?id=badatt_tidy-ui">
    <img
      src="https://sonarcloud.io/api/project_badges/measure?project=badatt_tidy-ui&metric=vulnerabilities"
      alt="Vulnerabilities"
    />
  </a>
  <a href="https://sonarcloud.io/summary/new_code?id=badatt_tidy-ui">
    <img src="https://sonarcloud.io/api/project_badges/measure?project=badatt_tidy-ui&metric=bugs" alt="Bugs" />
  </a>
  <a href="https://sonarcloud.io/summary/new_code?id=badatt_tidy-ui">
    <img
      src="https://sonarcloud.io/api/project_badges/measure?project=badatt_tidy-ui&metric=security_rating"
      alt="Security Rating"
    />
  </a>
  <a href="https://sonarcloud.io/summary/new_code?id=badatt_tidy-ui">
    <img
      src="https://sonarcloud.io/api/project_badges/measure?project=badatt_tidy-ui&metric=sqale_rating"
      alt="Maintainability Rating"
    />
  </a>
  <a href="https://sonarcloud.io/summary/new_code?id=badatt_tidy-ui">
    <img
      src="https://sonarcloud.io/api/project_badges/measure?project=badatt_tidy-ui&metric=code_smells"
      alt="Code Smells"
    />
  </a>
  <a href="https://sonarcloud.io/summary/new_code?id=badatt_tidy-ui">
    <img
      src="https://sonarcloud.io/api/project_badges/measure?project=badatt_tidy-ui&metric=ncloc"
      alt="Lines of Code"
    />
  </a>
  <a href="https://sonarcloud.io/summary/new_code?id=badatt_tidy-ui">
    <img
      src="https://sonarcloud.io/api/project_badges/measure?project=badatt_tidy-ui&metric=sqale_index"
      alt="Technical Debt"
    />
  </a>
  <a href="https://sonarcloud.io/summary/new_code?id=badatt_tidy-ui">
    <img
      src="https://sonarcloud.io/api/project_badges/measure?project=badatt_tidy-ui&metric=reliability_rating"
      alt="Reliability Rating"
    />
  </a>
  <a href="https://sonarcloud.io/summary/new_code?id=badatt_tidy-ui">
    <img
      src="https://sonarcloud.io/api/project_badges/measure?project=badatt_tidy-ui&metric=duplicated_lines_density"
      alt="Duplicated Lines (%)"
    />
  </a>
</p>


## Project Description

Tidy UI is a family of components necessary for your react application. The components are powered [styled-components](https://styled-components.com/), features include css-in-js, theming etc

## Build with

- react
- styled-components
- rollup


## Installation

```cmd
yarn add @tidy-ui/commons
```

```cmd
npm i @tidy-ui/commons
```

## Getting started

Here is a sample root level component that should enclose all tidy-ui components

```typescript
import React from 'react';
import { TidyUiProvider, orchidLight } from '@tidy-ui/commons';

function App() {
  return <TidyUiProvider theme={orchidLight}>...</TidyUiProvider>;
}

export default App;
```

Include required components with in the place holder `...`

