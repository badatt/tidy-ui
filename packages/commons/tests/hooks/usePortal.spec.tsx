/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { orchidLight } from '../../commons/src';
import { usePortal, TidyUiProvider } from '../../src';

describe('usePortal', () => {
  it('Should create Portal on body', () => {
    const TestComponent = () => {
      const { Portal } = usePortal({ container: document.body });
      return <Portal>portal</Portal>;
    };
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <TestComponent />
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Should create Portal on and element', () => {
    const TestComponent = () => {
      const { Portal } = usePortal({ container: () => document.getElementById('sample') });
      return <Portal>portal</Portal>;
    };
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <div id="sample"></div>
        <TestComponent />
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Should create Portal on an id', () => {
    const TestComponent = () => {
      const { Portal } = usePortal({ id: 'sample' });
      return <Portal>portal</Portal>;
    };
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <div id="sample"></div>
        <TestComponent />
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Should work on empty input', () => {
    const TestComponent = () => {
      const { Portal } = usePortal();
      return <Portal>portal</Portal>;
    };
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <div id="sample"></div>
        <TestComponent />
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
