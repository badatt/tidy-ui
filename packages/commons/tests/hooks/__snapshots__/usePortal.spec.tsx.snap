// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usePortal Not a valid dom 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="sample"
      />
    </div>
  </body>,
  "container": <div>
    <div
      id="sample"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`usePortal Should create Portal on an id 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="sample"
      />
    </div>
    portal
  </body>,
  "container": <div>
    <div
      id="sample"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`usePortal Should create Portal on and element 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="sample"
      />
    </div>
    portal
  </body>,
  "container": <div>
    <div
      id="sample"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`usePortal Should create Portal on body 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    portal
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`usePortal Should work on empty input 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="sample"
      />
    </div>
    portal
  </body>,
  "container": <div>
    <div
      id="sample"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
